src/django/api/views/sectors.py

```diff
@@ -1,76 +1,167 @@
 from api.constants import FacilityClaimStatuses
 from api.models.facility.facility_index import FacilityIndex
+from api.models.sector_group import SectorGroup
 from rest_framework.decorators import (
     api_view,
     throttle_classes,
 )
 from rest_framework.response import Response
 from django.db.models import F, Func
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
 
 from ..models.facility.facility_claim import FacilityClaim
 from ..models.facility.facility_list_item import FacilityListItem
 
 
-@api_view(['GET'])
-@throttle_classes([])
-def sectors(request):
-    """
-    Returns a list of suggested sectors submitted by contributors,
-    sorted alphabetically.
+@swagger_auto_schema(
+    method='get',
+    manual_parameters=[
+        openapi.Parameter(
+            'embed',
+            openapi.IN_QUERY,
+            description="If present, returns a flat list of sectors submitted "
+            "by a specific contributor.",
+            type=openapi.TYPE_BOOLEAN,
+        ),
+        openapi.Parameter(
+            'contributor',
+            openapi.IN_QUERY,
+            description="Contributor ID, required when 'embed' is present.",
+            type=openapi.TYPE_INTEGER,
+        ),
+        openapi.Parameter(
+            'grouped',
+            openapi.IN_QUERY,
+            description="If present, returns a grouped list of sectors by "
+            "their sector groups.",
+            type=openapi.TYPE_BOOLEAN,
+        ),
+    ],
+    operation_description="""
```

Comment:
The description should be moved to a separate file to make the definition of this method shorter. After moving it to a separate file, you can assign the value of the new variable to the `operation_description` param.